### PR TITLE
CRM: Adding a Back To List button on many edit and add new pages

### PR DESCRIPTION
**Pull request by coder-karen**
## Proposed changes:

* This PR adds a 'Back to List' button  to the top of the Edit Quote page. It also adds it to the Add New Quote page, to match the Add New and Edit Contact pages.
* Similarly, it does the same for the Add New and Edit Transaction, Invoice and Company pages (added later)

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

https://github.com/Automattic/zero-bs-crm/issues/2944

## Does this pull request change what data or activity we track or use?

No.

## Testing instructions:

* To test this, use the Jetpack Beta plugin on a test site with this branch applied or test locally with the branch checked out.
* Visit the 'Add New' Quote page (`/wp-admin/admin.php?page=zbs-add-edit&action=edit&zbstype=quote`), and make sure the Back to List button shows. Make sure the rest of the menu matches the screenshot below, and the links all work properly.
* Visit the 'Edit Quote' page for any existing quote (`/wp-admin/admin.php?page=zbs-add-edit&action=edit&zbstype=quote&zbsid={quoteid}`), and make sure the Back to List button shows. Make sure the rest of the menu matches the screenshot below, and the links all work properly.
* Visit the same two pages on trunk (or using the release candidate / bleeding edge if using the Beta plugin), and make sure the top navigation is the same as before minus the Back to List button.
* As well as as Quotes, check the Transactions, Invoices and Companies pages as well. The only difference should be the Back to List button.

Screenshots (just for Quotes, as an example):

**Before**

Add New Quote page:

Screenshot 2023-03-23 at 10 20 47

Edit Quote page:

Screenshot 2023-03-23 at 10 21 31

**After**

Add New Quote page:
Screenshot 2023-03-23 at 10 20 06

Edit Quote page:

Screenshot 2023-03-23 at 10 19 43

### CODE REVIEW

File: projects/plugins/crm/includes/ZeroBSCRM.FormatHelpers.php [hunk 1]
```diff
@@ -1074,23 +1071,41 @@ function zeroBSCRM_getObjNav( $id = -1, $key = '', $type = ZBS_TYPE_CONTACT ) {
...
+ * @param string $slug - The slug for the page.
+ * @return string $html - The HTML string.
+ */
+function zeroBSCRM_print_backtolist_html( $slug ) {
+	$html = '<span class="ui navigation-quick-links">';
+
+	$html .= '<a style="margin-right:6px;" href="' . jpcrm_esc_link( $slug ) . '" class="ui button mini was-inverted basic" id="back-to-list">' . esc_html( __( 'Back to List', 'zero-bs-crm' ) ) . '</a>';
+
+	return $html;
+}
```
Comment (gogdzl) [inline projects/plugins/crm/includes/ZeroBSCRM.FormatHelpers.php:1121]: Love this!

**Review by gogdzl — APPROVED** [review]
Looks good to me!
Thanks for cleaning up some of the old code.

File: projects/plugins/crm/includes/ZeroBSCRM.FormatHelpers.php (hunk 1 above)
Comment (coder-karen) [reply projects/plugins/crm/includes/ZeroBSCRM.FormatHelpers.php:1121]: 🥳

**Comment by coder-karen** [comment]
@gogdzl thanks! Before I merge, would it be better to do the same for all other similar issues in the one PR ([mentioned here](https://github.com/Automattic/zero-bs-crm/issues/2944#issuecomment-1480968875)), or would it be preferred to do these separately because of the epics they will be merged into?

Or at least the remaining issues could all be done together. How would that work with the project and the epics?

**Comment by gogdzl** [comment]
You can go ahead and do all in one PR.

**Comment by coder-karen** [comment]
Thanks @gogdzl - I've add a new commit with all changes in them now.

**Review by gogdzl — APPROVED** [review]

**Pull request merged**